nautobot_circuit_maintenance/handle_notifications/sources.py

```diff
@@ -193,9 +215,65 @@ def receive_notifications(
         self.close_session()
         return received_notifications
 
+    def validate_providers(self, logger: Job, notification_source: NotificationSource, since_txt: str) -> bool:
+        """Method to validate that the NotificationSource has attached Providers.
+
+        Args:
+            logger (Job): Job to use its logger
+            notification_source (NotificationSource): Notification Source to validate providers
+            since_txt (str): Date string to be used to log
+
+        Returns:
+            bool: True if there are relevant providers attached or False otherwise
+        """
+        providers_with_email = []
+        providers_without_email = []
+        if not notification_source.providers.all():
+            logger.log_warning(
+                message="Skipping this email account no providers were defined.",
+            )
+            return False
+
+        for provider in notification_source.providers.all():
+            provider_emails = ""
+            for custom_field, value in provider.get_custom_fields().items():
+                if custom_field.name == "emails_circuit_maintenances" and value:
+                    provider_emails = value
+                    break
```

Comment:
`provider.get_custom_fields()` returns and `OrderedDict` that doesn't work fine with `.get("emails_circuit_maintenance", "")`... but maybe it's me who is missing something 🤒

Reply:
Really? `OrderedDict` is a subclass of `dict`, it ought to support the same syntax.

Reply:
it actually works as you say and I expected, but the code was not behaving well... will double check to validate it
```py
>>> from collections import OrderedDict
>>> numbers = OrderedDict()
>>> numbers["one"] = 1
>>> numbers.get("one")
```

Reply:
@glennmatthews -> b1392c1

Reply:
Ah, okay, I hadn't remembered that `get_custom_fields()` returns a dict of `CustomField` objects. That makes sense then. The way you've coded it is fine; the alternative would be to use `provider.cf` (which is an ordinary key-value dictionary) instead of `provider.get_custom_fields()`, but either way is fine.